ceh-2000/gpu_guru
# Implement FedVAE decoder averaging and sampling according to number of samples

# Closes #43 

## Description
This ticket implements weighted averaging of the server decoder and weighted sampling to train the server decoder and classifier. 

## Testing
1. Pull down this branch `feature/43_tweak-kd`
2. Run `README.md` experiments for FedVAE and OneFedVAE.
3. Go to the code diffs and check that all necessary changes have been made to FedVAE AND OneFedVAE. Note that OneFedVAE uses the same user file as FedVAE, so you really just have to check the `train` method of ServerOneFedVAE.
4. Because MNIST is balanced, the number of targets that we generate for each class should be roughly the same. Check that this is the case.
